twine/settings.py

```diff
@@ -253,12 +253,20 @@ def _handle_authentication(self, username, password):
             username,
             self.repository_config
         )
-        self.password = utils.get_password(
-            self.repository_config['repository'],
-            self.username,
-            password,
-            self.repository_config,
-        )
+        if self.client_cert:
+            self.password = utils.get_password_no_prompt(
+                self.repository_config['repository'],
+                self.username,
+                password,
+                self.repository_config,
+            )
+        else:
+            self.password = utils.get_password(
+                self.repository_config['repository'],
```

Comment:
This code is repetitive. If this `if/else` block remains, I suggest instead selecting `get_password = utils.get_password if self.client_cert else utils.get_password_no_prompt` and then call the parameters with that.